homeassistant/components/tplink_omada/controller.py

```diff
@@ -1,10 +1,17 @@
 """Controller for sharing Omada API coordinators between platforms."""
 
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
 from tplink_omada_client import OmadaSiteClient
 from tplink_omada_client.devices import OmadaSwitch
 
 from homeassistant.core import HomeAssistant
 
+if TYPE_CHECKING:
+    from . import OmadaConfigEntry
+
 from .coordinator import (
```

Comment:
We also could move the type definition to `coordinator.py` like we have done in other PR's.

Reply:
At the beginning I had the same idea, but finally decided to only moved it, when it is direct and only related to the coordinator